internal/catalogmetadata/client/client.go

```diff
@@ -111,6 +115,24 @@ func (c *Client) PopulateCache(ctx context.Context, catalog *catalogd.ClusterCat
 	return c.cache.Put(catalog.Name, catalog.Status.ResolvedSource.Image.Ref, resp.Body, nil)
 }
 
+func logX509Error(err error, l logr.Logger) {
```

Comment:
Another option (which has the downside of losing structured output) is that we could implement an error wrapper with a custom `Error() string` method and put this logic there to essentially beef up the existing error message.

Reply:
Something to be considered.

Reply:
But I must admit, I like the structured log